modules/uhttp/client/client.go

```diff
@@ -21,115 +21,56 @@
 package client
 
 import (
-	"io"
 	"net/http"
-	"net/url"
-	"strings"
+	"time"
 
 	"go.uber.org/fx"
 	"go.uber.org/fx/auth"
-	"go.uber.org/fx/config"
-
-	"golang.org/x/net/context/ctxhttp"
-)
-
-var (
-	_serviceName string
+	"go.uber.org/fx/internal/fxcontext"
 )
 
-// Client wraps around a http client
-type Client struct {
-	*http.Client
-	info                auth.CreateAuthInfo
-	filters             []Filter
-	defaultFiltersAdded bool
-}
-
-// New creates a new instance of uhttp Client
-func New(info auth.CreateAuthInfo, client *http.Client, filters ...Filter) *Client {
-	_serviceName = info.Config().Get(config.ApplicationIDKey).AsString()
-	filters = append(filters, tracingFilter(), authenticationFilter(info))
-	return &Client{
-		Client:              client,
-		info:                info,
-		filters:             filters,
-		defaultFiltersAdded: true,
-	}
-}
-
-// Do is a context-aware, filter-enabled extension of Do() in http.Client
-func (c *Client) Do(ctx fx.Context, req *http.Request) (resp *http.Response, err error) {
-	filters := c.filters
-	if c.defaultFiltersAdded == false {
-		// TODO(anup): GFM-289 Update uhttp client to not return exported struct
-		filters = append(filters, tracingFilter(), authenticationFilter(c.info))
-		c.defaultFiltersAdded = true
+// New creates an http.Client that includes 2 extra filters: tracing and auth
+// they are going to be applied in following order: tracing, auth, remaining filters
+// and only if all of them passed the request is going to be send.
+func New(info auth.CreateAuthInfo, filters ...Filter) *http.Client {
+	defaultFilters := []Filter{tracingFilter(), authenticationFilter(info)}
+	defaultFilters = append(defaultFilters, filters...)
+	return &http.Client{
+		Transport: newExecutionChain(defaultFilters, http.DefaultTransport),
+		Timeout:   2 * time.Minute,
```

Comment:
Can we make this configurable? I know 2 minutes is a long time, but needs user modifiable.